### DISCUSSION TITLE
How to setup open each external link in new tab?

### DISCUSSION THREAD

**Discussion by mxcdh** [vercel/next.js#35702] [Help]
Hello

How I can set up in nextjs open each external link in new tab, without adding each time to link "_blank" etc? I want that this setup working also with Link, a href, and markdown in md files? 

Any ideas?

**Comment by jamesmosier** [comment]
There is no built-in way to do this in Next. `<Link>` components are meant for navigating to a page within your application without reloading the entire page (it's a client side transition). Aside from that, the only way to open a new tab without `target="_blank"` would be to use JavaScript and do `window.open(url, '_blank')`. There is no way to automate this for all `a` tags in Next without using custom JavaScript to listen to all clicks on the page:

```js
document.addEventListener(`click`, e => {
  const element = e.target.closest(`a`);
  const url = element.href;
  window.open(url, '_blank');
});
```

**Reply by mxcdh** [reply]
@jamesmosier  ok thanks, but where I should put this code?

**Reply by jamesmosier** [reply]
You could use the [`next/script`](https://nextjs.org/docs/basic-features/script) tag and put it in `_app.js` if you want it to run on all pages:

```js
<Script
  strategy="afterInteractive"
  dangerouslySetInnerHTML={{
    __html: `
    // custom javascript code here 
  `,
  }}
/>
```
